pkg/hubble/container/ring_test.go

```diff
@@ -503,20 +503,17 @@ func TestRingFunctionalityInParallel(t *testing.T) {
 		t.Errorf("lastWrite should be 0x0. Got %x", lastWrite)
 	}
 
-	entry, ok := r.read(lastWrite)
-	if !ok {
-		t.Errorf("Should be able to read position %x", lastWrite)
+	entry, err := r.read(lastWrite)
+	if err != nil {
+		t.Errorf("Should be able to read position %x, got %v", lastWrite, err)
 	}
 	if entry.Timestamp.Seconds != int64(0) {
 		t.Errorf("Read Event should be %+v, got %+v instead", &timestamp.Timestamp{Seconds: 0}, entry.Timestamp)
 	}
 	lastWrite--
-	entry, ok = r.read(lastWrite)
-	if !ok {
-		t.Errorf("Should be able to read position %x", lastWrite)
-	}
-	if entry != nil {
-		t.Errorf("Read Event should be %+v, got %+v instead", nil, entry)
```

Comment:
Reviewer note: This test relied on `read` returning `nil` values. With this PR, we never allow `nil` values to be returned from `read` (same as `readFrom`, which had this restriction already for a while)